action.yml

```diff
@@ -24,35 +23,70 @@ runs:
     - name: approve PR
       uses: actions/github-script@v6.3.3
       with:
+        github-token: ${{ steps.vault.outputs.GITHUB_MERGE_TOKEN }}
         script: |
           const opts = github.rest.pulls.listReviews.endpoint.merge({
             pull_number: context.payload.pull_request.number,
             owner: context.payload.repository.owner.login,
             repo: context.payload.repository.name,
           });
+
           const reviews = await github.paginate(opts);
+
           const ourReview = reviews.find(
             (review) =>
               review.state === "APPROVED" && review.user.login === "contentful-automation[bot]"
           );
+
           if (ourReview) {
             console.log(
               `The user "${ourReview.user.login}" has already approved and requested this PR is merged, exiting`
             );
-            return;
+          } else {
+            github.rest.pulls.createReview({
+              owner: context.payload.repository.owner.login,
+              repo: context.payload.repository.name,
+              pull_number: context.payload.pull_request.number,
+              event: 'APPROVE',
+              body: ''
+            })
           }
-          github.rest.pulls.createReview({
+    - name: Get merge type
+      id: merge-type
+      uses: actions/github-script@v6.3.3
+      with:
+        github-token: ${{ steps.vault.outputs.GITHUB_MERGE_TOKEN }}
+        script: |
+          const repo = await github.rest.repos.get({
             owner: context.payload.repository.owner.login,
             repo: context.payload.repository.name,
-            pull_number: context.payload.pull_request.number,
-            event: 'APPROVE',
-            body: ''
-          })
+          });
 
-        github-token: ${{ steps.vault.outputs.GITHUB_MERGE_TOKEN }}
+          const methods = {
+            squash: repo.data.allow_squash_merge,
+            merge: repo.data.allow_merge_commit,
+            rebase: repo.data.allow_rebase_merge,
+          };
+
+          console.log(methods)
+
+          const allowedMethods = Object.entries(methods)
+            .filter(([_, allowed]) => allowed)
+            .map(([method]) => method);
+
+          // just pick the first one
```

Comment:
shall we prefer squash?

Reply:
Good point - it _should_ do this anyway because of the ordering of keys in the object, but I'll switch to a `Map` to ensure this is guaranteed.